File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRightDifference = function(nums) {
      let leftSum = [];
  let rightSum = [];
  let arr = [];
  let leftRunSum = 0;
  let rightRunSum = 0;

  for (let i = 0, j = nums.length - 1; i < nums.length; i++, j--) {
    leftSum[i] = leftRunSum;
    leftRunSum += nums[i];

    rightSum[j] = rightRunSum;
    rightRunSum += nums[j];
  }

  for (let i = 0; i < nums.length; i++) {
    let diff = leftSum[i] - rightSum[i];

    if (diff > 0) {
      arr.push(diff);
    } else {
      arr.push(-diff);
    }
  }
  return arr;
};

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        left_sum = [0] + [sum(nums[: i + 1]) for i in range(len(nums) - 1)]
        right_sum = [sum(nums[i + 1 :]) for i in range(len(nums))]
        
        return [abs(left_sum[i] - right_sum[i]) for i in range(len(nums))]